run_querytns.py

```diff
@@ -0,0 +1,233 @@
+
+.. codeauthor:: Emir Karamehmetoglu <emir.k@phys.au.dk>
+.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+"""
+
+import argparse
+import logging
+from flows import api, load_config
+import requests
+import json
+from collections import OrderedDict
+from bs4 import BeautifulSoup
+import numpy as np
+from astropy.coordinates import SkyCoord
+import astropy.units as u
+from astropy.time import Time
+from datetime import datetime, timedelta
+import time
+
+#--------------------------------------------------------------------
+# TNS API FUNCTIONS
+# Pre-provided helper functions for the TNS API
+# Obtained from https://wis-tns.weizmann.ac.il/content/tns-getting-started
+# function for changing data to json format
+url_tns_api = "https://wis-tns.weizmann.ac.il/api/get"
+def format_to_json(source):
+    # change data to json format and return
+    parsed = json.loads(source,object_pairs_hook=OrderedDict)
+    result = json.dumps(parsed,indent=4)
+    return result
+
+# function for search obj
+def search(url,json_list):
+    try:
+        # url for search obj
+        search_url = url+'/search'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        search_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # search obj using request module
+        response = requests.post(search_url, files=search_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+
+# function for get obj
+def get_obj_tns(url,json_list):
+    try:
+        # url for get obj
+        get_url = url+'/object'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        get_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # get obj using request module
+        response = requests.post(get_url, files=get_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+
+# Get SN names from TNS html table
+def get_names(soup1):
+    names = soup1.find_all('td','cell-name')
+    names_list = []
+    for name in names:
+        if name.text.startswith('SN'):
+            names_list.append(name.text.replace(' ',''))
+    return names_list
+
+#--------------------------------------------------------------------
+if __name__ == '__main__':
+    # Parse command line arguments:
+    parser = argparse.ArgumentParser(description='Query TNS and upload to candidate marshal')
+    parser.add_argument('-d', '--debug', help='Print debug messages.', action='store_true')
+    parser.add_argument('-q', '--quiet', help='Only report warnings and errors.', action='store_true')
+    parser.add_argument('-z', '--zmax', type=float, help='Z max', default=0.105)
```

Comment:
Could you make a more helpful help-message than "zmax = Z max" and "zmin = Z min".... Is it "Maximum redshift"?